src/main/java/com/strandls/document/controllers/DocumentController.java

```diff
@@ -96,6 +100,58 @@ public Response getPong() {
 		return Response.status(Status.OK).entity("PONG").build();
 	}
 
+	@PUT
+	@Path(ApiConstants.DELETENAME + "/{nameId}")
+	@Consumes(MediaType.TEXT_PLAIN)
+	@Produces(MediaType.APPLICATION_JSON)
+
+	@ApiOperation(value = "delete scientific name detected for a document by gnfinder", notes = "returns the object with updated column", response = DocSciName[].class)
+	@ApiResponses(value = { @ApiResponse(code = 400, message = "unable to fetch the data", response = String.class) })
+
+	@ValidateUser
+	public Response deleteScientificName(@Context HttpServletRequest request, @PathParam("nameId") Long nameId) {
+		try {
+			DocSciName response = docService.updateScienticNametoIsDeleted(request, nameId);
+			return Response.status(Status.OK).entity(response).build();
+		} catch (Exception e) {
+			return Response.status(Status.BAD_REQUEST).entity(e.getMessage()).build();
+		}
+
+	}
+
+	@GET
+	@Path(ApiConstants.GNRD)
+	@Consumes(MediaType.TEXT_PLAIN)
+	@Produces(MediaType.APPLICATION_JSON)
+
+	@ApiOperation(value = "detects scientific names in pdf files through gnfinder", notes = "returns the scientific names details detected by gnfinder", response = GNFinderResponseMap.class)
+	@ApiResponses(value = { @ApiResponse(code = 400, message = "unable to fetch the data", response = String.class) })
+
+	public Response parsePdfWithGNFinder(@QueryParam("filePath") String filePath,
+			@QueryParam("documentId") Long documentId) {
+		GNFinderResponseMap response = docService.parsePdfWithGNFinder(filePath, documentId);
+		return Response.status(Status.OK).entity(response).build();
+	}
+
+	@GET
+	@Path(ApiConstants.NAMES + "/{documentId}")
+	@Consumes(MediaType.TEXT_PLAIN)
+	@Produces(MediaType.APPLICATION_JSON)
+
+	@ApiOperation(value = "fetch the scientific names in pdf documents", notes = "returns the scientific names detected by gnfinder", response = DocSciName[].class)
+	@ApiResponses(value = { @ApiResponse(code = 400, message = "unable to fetch the data", response = String.class) })
+
+	public Response findNamesInDocument(@PathParam("documentId") Long documentId, @QueryParam("offset") int offset) {
```

Comment:
Don't use primitive type int, its doesn't have the capability to hold null values, use wrapper class

Reply:
Corrected.